Implement trades_history for Kraken and improve the Trade Class

gagdiez opened the pull request:
In this commit I implemented the trades_history
for the kraken exchange. A test for the method
was also added. An important change made during
this commit is the addition of the parameter
time to the class Trade. This not only allows
to simplify the implementation of trades_history
but also helps to better characterize a trade.

gagdiez committed:
Implement trades_history for Kraken and improve the Trade Class

In this commit I implemented the trades_history
for the kraken exchange. A test for the method
was also added. An important change made during
this commit is the addition of the parameter
time to the class Trader. This not only allows
to simplify the implementation of trades_history
but also helps to better characterize a trade.

gagdiez commented:
Now, a trade is a tuple (asset_pair, price, volume). In this pull request I made a change to add also "time". This makes sense to me since it's not the same to trade 1 ETH for 10E today than what it was two months ago. Today is a bad trade, two months ago it was natural. Also, this helps to create a nice implementation for trades_history.

sdeslauriers commented on exchanges/kraken.py:
```diff
@@ -19,16 +20,46 @@ def latest_trade(self, asset_pair):
+            # The second time we make a query, we can use the
+            # 'last ID' of the previous query as the parameter 'since'.
+            # This is recommended in the api's documentation
+            response = self._public_query('Trades', {'pair':asset_pair,
```
Do public data queries have a call rate limit? If so, we might want to add a delay to not get banned.

gagdiez replied:
For what I understand, only private queries have a rate limit. Anyway, with the current implementation we cannot hit the limit. Each call to response is really slow (since the connection is not persistent).

sdeslauriers replied:
Yeah, it is not clear from the documentation how they measure the call rate. API key based or IP or ...
If they are monitored, I think it could still happen. Querying trades increases the counter by 2 and it only decreases by 1 every 3 seconds. While slow, we still get queries in < 1 sec.
Keep it like this and if we notice we get banned, we could add an optional param for delays.

gagdiez replied:
I'll do some local test (which I'll not add to the repo) to try that. I'll make a query for one month of data, and leave it running, checking if at some point one requests takes more than 15min.

sdeslauriers commented on exchanges/tests/test_kraken.py:
```diff
@@ -50,3 +50,47 @@ def test_latest_ticker(self):
+        # in a web browser we can create a specific test with only
+        # 3 orders between 'since' and 'until'
+        until = 1470092421.1177
+        trades_history = kraken.trades_history(asset, since, until)
```
Here, could you check that you have the correct prices and volumes? This is not verified anywhere in the tests.

gagdiez replied:
good point, I'll do it now

gagdiez committed:
Improve test case for trades_history

In this commit I improved one of the test cases for trades_history.
Since we can access data throught a web browser, I can check for
the exact prices and volumes that we should receive when passing
some especific parameters.

sdeslauriers reviewed (approved):
Looks good to me.

gagdiez commented:
If I PR, you merge.
